### ISSUE TITLE
How to create a vararg function with `deflua`

### ISSUE THREAD

**Issue by fermuch** [tv-labs/lua#27]
If we want to re implement `print(...)` via `deflua`, what would be the recommended way to do it?

**Comment by davydog187** [comment]
Hello @fermuch, you've found a gap in the documentation! I just pushed #29 and tagged a new release.

You can see how to implement variadic functions via `deflua` [in this example](https://hexdocs.pm/lua/Lua.API.html#deflua/3-variadic-functions).

In short, you can use the `@variadic true` module attribute to annotate the function as variadic. A simple version of `print` might look like this

```elixir
def MyGlobalFuncs do
  use Lua.API

  @variadiac true
  deflua print(args) do
    IO.puts(Enum.join(args, " "))
  end
end
```

You'd then need to load this into your script with `Lua.load_api`. There's plenty of examples in the docs on that.

Thanks for your interest, and let me know if I can help with anything else!

**Comment by fermuch** [comment]
Thank you!

**Comment by davydog187** [comment]
I pushed too fast and didn't catch a spelling error. Make sure you use "@variadic" not "@variadiac"

**Comment by fermuch** [comment]
I think there is a problem with the implementation when using state.

Here is an example:

```elixir
defmodule MyGlobalFuncs do
  use Lua.API

  @variadic true
  deflua print(args), state do
    IO.puts(Enum.join(args, " "))

    {[], state}
  end
end

Lua.new() |> Lua.load_api(MyGlobalFuncs) |> Lua.eval!("print('foo', 'bar')")
```

But it fails with:

```elixir
** (Lua.RuntimeException) Lua runtime error: Elixir.MyGlobalFuncs.print(_, _, _)
    MyGlobalFuncs.print("foo", "bar", #Lua<>)
```

**Comment by davydog187** [comment]
Hmm, ok maybe I didn't test this edge case. I'll take a look

**Comment by davydog187** [comment]
Fwiw in this example there's no reason to pull in state

**Comment by fermuch** [comment]
> Fwiw in this example there's no reason to pull in state

Yeah, I know. My real print function uses the state to get the current executing line. I've tried to simplify the code to make it iex-pastable to see it raise the error immediately :stuck_out_tongue_winking_eye:

**Comment by davydog187** [comment]
@fermuch this issue has now been fixed and I've published a new version v0.0.14

Let me know if you run into any other issues
